### DISCUSSION TITLE
As to the DSA part, why not use BFS?

### DISCUSSION THREAD

**Discussion by bladerunnersde** [LordMoMA/maze#2] [Q&A] [answered]
DFS has to be implemented with recursions, why you can simply implement BFS with a queue.

**Comment by LordMoMA** [comment, marked as answer]
If the graph becomes too wide, it will be time-consuming to reach the the bottom exit. DFS is optimal here.

**Reply by bladerunnersde** [reply]
thank you so much, i learned a lot.
